### PR TITLE
Update Packet CCM to v1.0.0

### LINKED ISSUE #750: Nodes can not be initialized by packet CCM
Packet CCM logging errors:

```
ERROR: logging before flag.Parse: E1203 13:36:21.266478       1 node_controller.go:417] NodeAddress: Error fetching by providerID: providerID cannot be empty string Error fetching by NodeName: instance not found
```

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

* Update Packet CCM to v1.0.0
* Automatically create a secret with credentials and configuration needed for the CCM
* Kubernetes 1.15.0+ is now required for Packet

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #750

**Does this PR introduce a user-facing change?**:
```release-note
Update Packet CCM to v1.0.0
Automatically create a secret with credentials and configuration needed for the CCM
Kubernetes 1.15.0+ is now required for Packet
```

### CODE REVIEW

**Comment by xmudrii** [comment]
/hold cancel
the PR is tested and the doc change is not required anymore

File: pkg/templates/externalccm/packet.go [hunk 1]
```diff
@@ -34,9 +36,18 @@ import (
...
 
 const (
-	packetImage          = "packethost/packet-ccm:v0.0.4"
+	packetImage          = "packethost/packet-ccm:v1.0.0"
 	packetSAName         = "cloud-controller-manager"
 	packetDeploymentName = "packet-cloud-controller-manager"
+
+	packetCloudSASecretName = "packet-cloud-config"
+	//nolint:gosec
+	packetCloudSASecretData = `
```
Comment (kron4eg) [inline pkg/templates/externalccm/packet.go:45]: Pretty sure that small map[string]string or even a struct with fields will do the trick without templates.

Comment (xmudrii) [reply pkg/templates/externalccm/packet.go:45]: I'm don't think `map[string]string` would work, but a struct with fields is a great idea.

Comment (xmudrii) [reply pkg/templates/externalccm/packet.go]: @kron4eg Fixed, PTAL.

**Pull request merged**